cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/KafkaSampleStore.java

```diff
@@ -170,22 +214,57 @@ private void ensureTopicsCreated(Map<String, ?> config) {
                               numberOfBrokersInCluster) : _sampleStoreTopicReplicationFactor;
 
       ensureTopicCreated(zkUtils, topics.keySet(), _partitionMetricSampleStoreTopic, partitionSampleRetentionMs,
-                         replicationFactor);
+                         replicationFactor, _partitionSampleStoreTopicPartitionCount);
       ensureTopicCreated(zkUtils, topics.keySet(), _brokerMetricSampleStoreTopic, brokerSampleRetentionMs,
-                         replicationFactor);
+                         replicationFactor, _brokerSampleStoreTopicPartitionCount);
     } finally {
       KafkaCruiseControlUtils.closeZkUtilsWithTimeout(zkUtils, 10000);
     }
   }
 
-  private void ensureTopicCreated(ZkUtils zkUtils, Set<String> allTopics, String topic, long retentionMs, int replicationFactor) {
+  private void increaseTopicReplicaFactor(ZkUtils zkUtils, MetadataResponse.TopicMetadata topicMetadata, int replicationFactor,
+      String topic, Properties props) {
+    List<Integer> brokers = new ArrayList<>();
+    JavaConversions.seqAsJavaList(AdminUtils.getBrokerMetadatas(zkUtils, RackAwareMode.Safe$.MODULE$, Option.empty()))
+    .stream().forEach(bm -> brokers.add(bm.id()));
+    if (replicationFactor > brokers.size()) {
+      throw new RuntimeException("Unable to increase topic " + topic + " replica factor to " + replicationFactor +
+                                 " since there are only " + brokers.size() + " brokers in the cluster.");
+    }
+    scala.collection.Map<Object, Seq<Object>> newReplicaAssignment = new scala.collection.mutable.HashMap<>();
+    int cursor = 0;
+    for (MetadataResponse.PartitionMetadata pm : topicMetadata.partitionMetadata()) {
+      List<Object> newAssignedReplica = new ArrayList<>();
+      // Make sure the current replicas are in new replica list.
+      pm.replicas().forEach(node -> newAssignedReplica.add(node.id()));
+      // Add new replica to partition in round-robin way.
+      while (newAssignedReplica.size() < replicationFactor) {
+        if (!newAssignedReplica.contains(brokers.get(cursor))) {
+          newAssignedReplica.add(brokers.get(cursor));
+        }
+        cursor = (cursor + 1) % brokers.size();
+      }
+      ((scala.collection.mutable.HashMap<Object, Seq<Object>>) newReplicaAssignment).put(pm.partition(),
```

Comment:
Would the following tip work:
`Cast may be removed by changing the type of 'newReplicaAssignment' to 'HashMap<Object, Seq<Object>>'`